histFactory/templates/Plot.tpl

```diff
@@ -1,6 +1,6 @@
         __cut = ({{CUT}});
         if (__cut) {
-            __weight = ({{WEIGHT}}) * __sample_weight;
+            __weight = runOnMC ? ( ({{WEIGHT}}) * __sample_weight ) : 1;
```

Comment:
I'd prefer an old fashioned `if` here if you don't mind. We already put a lot of stress on the compiler with our crazy weight expressions ;)